src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -17,14 +17,19 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	implements ToAPIApplicationCommandOptions
 {
 	public choices?: APIApplicationCommandOptionChoice[];
+	public autocomplete?: boolean;
 
 	/**
 	 * Adds a choice for this option
 	 *
 	 * @param name The name of the choice
 	 * @param value The value of the choice
 	 */
-	public addChoice(name: string, value: T) {
+	public addChoice(name: string, value: T): Omit<this, 'setAutocomplete'> {
+		if (typeof this.autocomplete !== 'undefined') {
```

Comment:
If autocomplete is set to false, can't you still send choices?

Reply:
Since you can set choices when autocomplete is false, I think the only way to make the types work here is to refactor the method to `enableAutocomplete()` that only allows you to set the `autocomplete` field to `true`.

Reply:
We shouldn't do any defaulting of that nature. While redundant, we should let users set whatever they want that fits the parameter type. I guess for simplicity sake just check if this.autocomplete, since if it's true it's invalid, but if it's false it's fine

Reply:
I agree with suggestions on validation. I was talking more about the return type of `setAutocomplete`, if you provide `false` to this method it still returns `Omit<this, 'addChoice' | 'addChoices'>`, which is incorrect because when`autocomplete` is `false` you can still set choices.

Now that I think of it, you can provide an overload to solve this issue:
```ts
setAutocomplete(autocomplete: false):  Omit<this, 'setAutocomplete'>
```